extensions/blocks/podcast-player/podcast-player.js

```diff
@@ -0,0 +1,372 @@
+		if ( this.currentTrack === track ) {
+			this.togglePlayPause();
+			return;
+		}
+
+		// Something else is playing.
+		if ( this.currentTrack !== -1 ) {
+			this.stop();
+		}
+
+		// Load a new track.
+		this.loadAndPlay( track );
+	}
+
+	/**
+	 * Set a player state classname.
+	 * @private
+	 * @param {string} state Player state.
+	 */
+	setPlayerState( state ) {
+		this.playerState = state;
+		this.block.classList.remove( STATE_ERROR, STATE_PAUSED, STATE_PLAYING );
+		this.block.classList.add( state );
+	}
+
+	/**
+	 * Load audio from the track, start playing.
+	 * @private
+	 * @param {number} track The track number
+	 */
+	loadAndPlay( track ) {
+		const trackData = this.getTrack( track );
+		if ( ! trackData ) {
+			return;
+		}
+		this.audio.src = trackData.src;
+		this.currentTrack = track;
+		this.play();
+		this.setTrackState( track, true );
+	}
+
+	/**
+	 * Visually mark track active or inactive.
+	 * @private
+	 * @param {number} track The track number
+	 * @param {boolean} isActive True if track is active
+	 * @param {boolean} isError True if track couldn't be played
+	 */
+	setTrackState( track, isActive, isError = false ) {
+		// Mark track element as active/inactive.
+		const el = this.trackElements[ track ];
+
+		// Active state classname.
+		if ( isActive ) {
+			el.classList.add( 'is-active' );
+		} else {
+			el.classList.remove( 'is-active' );
+		}
+
+		// Maintain pressed state.
+		el.querySelector( '[aria-pressed]' ).setAttribute( 'aria-pressed', JSON.stringify( isActive ) );
+
+		// Add or remove error.
+		const errorEl = el.querySelector( '.podcast-player__episode-error' );
+		if ( isError && ! errorEl ) {
+			const trackData = this.getTrack( track );
+			el.appendChild( this.renderErrorWithLink( trackData && trackData.link ) );
+		} else if ( ! isError && errorEl ) {
+			errorEl.remove();
+		}
+
+		const iconEl = el.querySelector( '.podcast-player__episode-status-icon' );
+		if ( iconEl ) {
+			if ( isActive ) {
+				if ( isError ) {
+					iconEl.innerHTML = errorIcon;
+				} else {
+					iconEl.innerHTML = soundIcon;
+				}
+			} else {
+				iconEl.innerHTML = '';
+			}
+		}
+	}
+
+	/**
+	 * Construct error markup.
+	 * @private
+	 * @param {string} link Optional URL to use as a fallback link
+	 * @returns {Element} Error DOM Element
+	 */
+	renderErrorWithLink( link ) {
+		const errorEl = document.createElement( 'div' );
+		errorEl.classList.add( 'podcast-player__episode-error' );
+		errorEl.appendChild( new Text( __( 'Episode unavailable', 'jetpack' ) ) );
+
+		if ( link ) {
+			const linkEl = document.createElement( 'a' );
+			linkEl.rel = 'noopener noreferrer nofollow';
+			linkEl.target = '_blank';
+			linkEl.href = link;
+			linkEl.innerText = __( 'Open in a new tab', 'jetpack' );
+
+			const spanEl = document.createElement( 'span' );
+			spanEl.appendChild( new Text( ' (' ) );
+			spanEl.appendChild( linkEl );
+			spanEl.appendChild( new Text( ')' ) );
+			errorEl.appendChild( spanEl );
```

Comment:
Also not blocking, but is this `span` necessary? Is it just for the `(` parentheses? Can we do `linkEl.innerText = __( '(Open in a new tab)', 'jetpack' );` instead?

Reply:
It would change the visual outcome of the element by adding underline below the parens:

| with `span`  | without `span` |
| ------------- | ------------- |
| <img width="128" alt="Screenshot 2020-03-19 21 19 46" src="https://user-images.githubusercontent.com/1451471/77111469-b3247080-6a27-11ea-96f4-8c100a74115e.png"> | <img width="134" alt="Screenshot 2020-03-19 21 20 20" src="https://user-images.githubusercontent.com/1451471/77111487-bb7cab80-6a27-11ea-867c-06bbf7ca8ae2.png"> |

Barely noticeable, but makes the difference! :D

Reply:
Idea: can we have a simple link and add parentheses using pseudo elements and also style them?

Reply:
I was wondering the same thing. Thanks for trying out the pseudo elements!